src/main/java/edu/hw3/task2/Clusterizer.java

```
@@ -0,0 +1,48 @@
+package edu.hw3.task2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Clusterizer {
+    private static final String CORRECT_REGEX = "[\\(\\)]*";
+
+    private Clusterizer() {
+        throw new IllegalStateException();
+    }
+
+    public static List<String> clusterize(String braces) {
+        if (!isCorrectInput(braces)) {
+            return null;
```

Лучше кидать исключение. Тогда вызывающий сразу получит ошибку, а не где-то дальше в своем флоу получит NPE. Принцип fail fast.